### PR TITLE
PWA: Action service migrated to ampdoc service

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #3742

### CODE REVIEW

File: src/service/action-impl.js [hunk 1]
```diff
@@ -87,17 +87,17 @@ export class ActionInvocation {
...
+  constructor(ampdoc) {
+    /** @const {!./ampdoc-impl.AmpDoc} */
+    this.ampdoc = ampdoc;
 
     /** @const @private {!Object<string, function(!ActionInvocation)>} */
     this.globalMethodHandlers_ = {};
 
     /** @private {!./vsync-impl.Vsync} */
-    this.vsync_ = vsyncFor(this.win);
+    this.vsync_ = vsyncFor(ampdoc.getWin());
```
Comment (cramforce) [inline src/service/action-impl.js]: Might be worth considering support passing AmpDoc to these factories.

**Comment by cramforce** [comment]
LGTM

File: src/service/action-impl.js (hunk 1 above)
Comment (dvoytenko) [inline src/service/action-impl.js]: So, for the time being, at least until most of migration is done, I'd really like to make a clear line between a window service and a ampdoc service, so I'd like to keep APIs as specific as possible w.r.t. arguments.

Comment (cramforce) [inline src/service/action-impl.js]: K

**Pull request merged**